userspace/libsinsp/container.cpp

```diff
@@ -273,21 +354,17 @@ bool sinsp_container_engine_docker::parse_docker(sinsp_container_manager* manage
 		string net_mode = hconfig_obj["NetworkMode"].asString();
 		if(strncmp(net_mode.c_str(), "container:", strlen("container:")) == 0)
```

Comment:
In the worst case, you might actually have a chain of dependent lookups, right? container a has network config that refers to container b, which refers to container c. This might be another way in which a state machine would help, because you could unwind the chain and return complete results as the dependent images have container information.

Reply:
It's probably possible and doesn't require much work, I will add support for it

Reply:
We chatted offline and the dependent lookup is used *only* to get the correct ip address to associate with the original container. That eliminates some of the cases I was thinking of, where the dependent container lookup would result in a PPME_CONTAINER_JSON event, etc. This makes sense now.